sdk/core/src/main/java/com/klaviyo/coresdk/model/SharedPreferencesDataStore.kt

```diff
@@ -0,0 +1,54 @@
+package com.klaviyo.coresdk.model
+
+import android.content.Context
+import android.content.SharedPreferences
+import com.klaviyo.coresdk.Klaviyo
+
+internal object SharedPreferencesDataStore : DataStore {
+    internal const val KLAVIYO_PREFS_NAME = "KlaviyoSDKPreferences"
+
+    /**
+     * Opens the Klaviyo SDK's shared preferences file
+     *
+     * @return The Klaviyo SDK's shared preferences opened in private mode
+     */
+    private fun openSharedPreferences(): SharedPreferences {
+        return Klaviyo.Registry.config.applicationContext.getSharedPreferences(
+            KLAVIYO_PREFS_NAME,
+            Context.MODE_PRIVATE
+        )
+    }
+
+    /**
+     * Opens the shared preferences and writes a given key/value pair
+     *
+     * The write operation is performed async, but has no callback
+     *
+     * @param key The identifying key that the value being written will go by
+     * @param value The value that we are writing to the shared preferences
+     */
+    override fun store(key: String, value: String) {
+        openSharedPreferences()
+            .edit()
+            .putString(key, value)
```

Comment:
Any need to mention that this obviously is only for pushing string values into sharedpreferences? Are there any cases where we may be pushing non-string data into our storage?

Reply:
Could do! The workaround is just to json encode of course, which is what I'm doing with the network request queue.